### PR TITLE
Add initial version of mcmc

### PR DESCRIPTION
**Pull request by mdml**
Add a simplified version of the CoMEt MCMC algorithm. Users can now use the MCMC to sample _multiple_ exclusive sets of mutations simultaneously, or enumerate all sets of a certain size. They do this using the `find_exclusive_sets.py` script (formerly known as `compute_exclusivity.py`).

I tested this a few times and the output seems reasonable, though perhaps we should test on the same data as was used by CoMEt to compare.

### CODE REVIEW

File: weighted_exclusivity_test/constants.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 PTOL = 10**-3
 
 # Set sizes implemented
-SET_SIZES_IMPLEMENTED = set([2, 3])
+WEIGHTED_SADDLEPOINT_SET_SIZES_IMPLEMENTED = set([2, 3])
```
Comment (matthewreyna) [inline weighted_exclusivity_test/constants.py:7]: In theory, we can compute the weighted saddlepoint for larger set sizes, too.

File: weighted_exclusivity_test/exclusivity_tests.py [hunk 2]
```diff
@@ -22,13 +23,19 @@ def weighted_test(t, x, p, method=EXACT, verbose=0):
     assert(all(a<=len(b) for a, b in zip(x, p)))
     # Check that the number of mutually exclusive mutations is not greater than the total number of mutations.
     assert(t<=sum(x))
+    #Check that we've implemented the given set size with the exact test
+    if method == EXACT:
+        assert( len(x) in WEIGHTED_SADDLEPOINT_SET_SIZES_IMPLEMENTED )
```
Comment (matthewreyna) [inline weighted_exclusivity_test/exclusivity_tests.py:28]: I see.  Should we rename this to `WEIGHTED_EXACT_SET_SIZES_IMPLEMENTED`?

File: weighted_exclusivity_test/constants.py (hunk 1 above)
Comment (mdml) [inline weighted_exclusivity_test/constants.py:7]: Yes, it should be `WEIGHTED_EXACT_SET_SIZES_IMPLEMENTED` instead.

File: weighted_exclusivity_test/io.py [hunk 3]
```diff
@@ -79,4 +83,37 @@ def output_table(args, setToPval, setToRuntime, setToFDR, setToObs, json_format=
...
+                      setsToFreq=dict( (' '.join([ ','.join(sorted(M)) for M in sets ]), freq) for sets, freq in setsToFreq.iteritems() ))
+        with open(args.output_prefix + '.json', 'w') as OUT:
+            json.dump( output, OUT )
+    else:
+        # Output a gene set file
+        with open(args.output_prefix + '-sampled-collections.tsv', 'w') as OUT:
+            rows = []
+            for sets, freq in setsToFreq.iteritems():
+                row = [ ' '.join([ ','.join(M) for M in sets ]), freq ]
+                row.append( sum( -np.log10(setToPval[M] ** args.alpha) for M in sets ))
```
Comment (matthewreyna) [inline weighted_exclusivity_test/io.py:101]: Replace with `row.append( -args.alpha * sum( np.log10(setToPval[M]) for M in sets ) )`.

**Pull request merged**